### ISSUE TITLE
[FEA] Implement TF compatible version of Novograd optimizer that supports sparse grads

### ISSUE THREAD

**Issue by rdipietro** [NVIDIA/NVTabular#122]
The Novograd optimizer has been shown to work well with large batch sizes.

As an example related to NVTabular, see https://github.com/NVIDIA/NVTabular/pull/9#issuecomment-649115880

However we weren't able to find a TF compatible implementation of Novograd that supports sparse gradients, which is required for use with embedding tables, and so it would be great to have such a version implemented.

Versions that we experimented with (without luck) include

- https://github.com/NVIDIA/OpenSeq2Seq/blob/master/open_seq2seq/optimizers/novograd.py
- https://www.tensorflow.org/addons/api_docs/python/tfa/optimizers/NovoGrad
- https://github.com/titu1994/keras_novograd
